### PR TITLE
[#16] Backend for getting survey list

### LINKED ISSUE #16: [Backend] As a user, I can see the list of surveys
## Why

App must call backend API to fetch list of surveys

## Acceptance Criteria

- Make post request to `https://survey-api.nimblehq.co/api/v1/surveys` and handle response
- If successful, return the list of surveys
- if unsuccessful, return the error from API

## Resources

- [API](https://documenter.getpostman.com/view/11835486/UVsPNPfU#064357a5-4625-4ef7-b042-0d1f7e2f2acb)

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes #16 

## What happened 👀

- Implemented `SurveyRepository`, `SurveyService`, `GetSurveysUseCase` and so on
- Added unit test for `GetSurveysUseCase`

## Insight 📝

- Updated the `DioProvider` to support the authorization header API.
- Add stack_trace dependency to get readability log since I faced the same issue with [this thread](https://nimble-co.slack.com/archives/C01NKCFNE6R/p1681266839179309).

## Proof Of Work 📹

- After testing, it calls `Get Surveys` service successfully:

Screenshot 2566-04-20 at 9 13 11 PM

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM, please check the CI status 🙏

File: lib/main.dart [hunk 1]
```diff
@@ -35,12 +42,12 @@ class MyApp extends StatelessWidget {
       GoRoute(
         path: routePathRootScreen,
         builder: (BuildContext context, GoRouterState state) =>
-            const LoginScreen(),
+        const LoginScreen(),
```
Comment (ryan-conway) [inline lib/main.dart]: Shall we revert this indentation change? Looks like it is causing the CI to fail 😄

File: lib/main.dart [hunk 2]
```diff
@@ -35,12 +42,12 @@ class MyApp extends StatelessWidget {
...
         routes: [
           GoRoute(
             path: routePathSecondScreen,
             builder: (BuildContext context, GoRouterState state) =>
-                const SecondScreen(),
+            const SecondScreen(),
```
Comment (ryan-conway) [inline lib/main.dart]: This one too 😄

**Review by ryan-conway — COMMENTED** [review]
Just two comments, the rest LGTM 🚀

**Comment by lydiasama** [comment]
@markgravity @ryan-conway I've pushed some update related the backend function. We should use the `async` and `await` in the repository. Can I get review from you again please? 🙏🏻

**Comment by lydiasama** [comment]
@markgravity @ryan-conway All issues have been fixed. 🎉  We will get rid of the `async` and `await` at the repository since it creates another exception binding issue. 😢

**Review by ryan-conway — APPROVED** [review]
🚀

**Pull request merged**